design/one-pager-stack-status-spec.md

```diff
@@ -0,0 +1,82 @@
+# Stack UI Schema Status Spec
+
+- Owner: Steven Rathbauer (@rathpc)
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Proposal
+
+[crossplaneio/crossplane#929](https://github.com/crossplaneio/crossplane/issues/929)
+
+## Problem
+
+As of right now there is no defined spec to be able to drive status information for a given stack or resource from a GUI perspective. It would be useful to add keys to the existing `ui-schema.yaml` file that can be used for this purpose.
+
+We can leverage the types from the array of **conditions** within the **conditionedStatus** key defined in **status**.
```

Comment:
Are conditions a common pattern in Kubernetes? (My initial research indicates that it [probably is](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions)). If so, we may be able to just define a default behavior that doesn't require any configuration. And then we could offer configurable behavior for people who want to override the default behavior.

Reply:
kubernetes has a general concept of conditions for status, but the implementation around `conditionedStatus` and query paths like what's found below are a crossplane specific concept. 

```
status.conditionedStatus.conditions[?(@.type=='${statusType}')]
```

This assumption and approach that everything has a `conditionedStatus` field would not hold true for arbitrary types defined by arbitrary stacks.  Have we considered a more general approach here?  Could the `ui-schema.yaml` allow the author to define a full query path instead of just a key name that gets injected into a hard-coded query path that assumes everything is using `conditionedStatus`?

Reply:
Those are both good points.

I was originally trying to limit the amount of configuration necessary while still having a valuable result. It seems that unless there is a concrete standard we may not be able to accommodate a "simple" config.

I have some thoughts on a solution but I need to sort them out. Let me do some thinking and get back to this.

Reply:
@suskin and @jbw976 

What do both of you think about this?

I think we can be flexible with this array and accept either a full path **OR** just a condition type string. So if an author is abiding by the crossplane specific concept, they have less work and can just pass the types as strings in the array, however it can also accept an array of full paths.

The only caveat here would be to strictly enforce if a path is sent, that the object it ultimately resolves to, follows the structure of a **[podcondition](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#podcondition-v1-core)**. This will ensure that all of the properties we expect within that object to be able to leverage from a GUI perspective can still be accessible.

Does that make sense? I would be happy to elaborate if needed.